idaes/gas_solid_contactors/unit_models/moving_bed.py

```diff
@@ -1177,15 +1246,39 @@ def initialize(blk, gas_phase_state_args=None, solid_phase_state_args=None,
             init_log.info('Initialize Energy Balances')
             with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
                 results = opt.solve(blk, tee=slc.tee)
-            if check_optimal_termination(results):
+            if results.solver.termination_condition \
+                    == TerminationCondition.optimal:
                 init_log.info_high(
                     "Initialization Step 4 {}.".format(
                             idaeslog.condition(results))
                             )
             else:
-                _log.warning('{} Initialisation Step 4 Failed.'
+                _log.warning('{} Initialization Step 4 Failed.'
                              .format(blk.name))
 
+    def calculate_scaling_factors(self):
+        super().calculate_scaling_factors()
+
+        if hasattr(self, "gas_phase_config_pressure_drop"):
```

Comment:
At a glance, this is not a lot of constraints being scaled. Make sure you scale every constraint added at the unit model level here.

Reply:
I originally added constraints under this method which previously had hard-coded factors, keeping @chineduobiora 's scaling factors. Would you recommend explicitly setting remaining constraint scaling factors to 1E0?

Reply:
I would recommend all other constraint have methods that determine a good scaling factor - you can specify a default scaling factor of 1 in the absence of more information, but always look for something better.

Reply:
My recommendation is to scale the equation based on the scale factors of the variables in the equation i.e.
`f: 	a*b = c*d.`

The constraint, f, can be scaled either with the LHS or the RHS i.e.
`sf1 = iscale.get_scaling_factor(a)`
`sf2 = iscale.get_scaling_factor(b)`
`iscale.constraint_scaling_transform(c, sf1 * sf2, overwrite=False)`

If the variables a and b do not have scale factors then they can be set from the equations they are derived from i.e.
`a = e/f`

`sf1 = iscale.get_scaling_factor(e)`
`sf2 = iscale.get_scaling_factor(f)`
`iscale.set_scaling_factor(v, sf1 * sf2)`

or a factor can be set if they are not derived from any equation i.e.
`iscale.set_scaling_factor(v, 1e1)`

In this way the scale factors for the constraints will end up being mostly derived from the scale factors that have been set in the property packages, and the factors of other geometric variables like the diameter and length of the reactor.

Reach out to me if it’s unclear and we can schedule a meeting to go through an example I have for the BFB.